### ISSUE TITLE
config level redirect with url params

### ISSUE THREAD

**Issue by airtonix** [inca/voie#10]
I'd like to setup a root state that redirects to to another state that requires params: 

```coffeescript
app.add 'root',
  path: '/'
  component: RootView
  redirect: name: 'form', param: id: 1

app.add 'form'
  parent: 'root'
  path: 'form/:id'
  component: FormView
``` 

Looking through the documentation, it looks like Voie doesn't support the following: 
```coffeescript
...
  redirect: name: 'form', param: id: 1
...
```

Any plans or a need for this?

perhaps you can suggest another way I can redirect users to the first form step when they hit `/`?

**Comment by inca** [comment]
Yep, redirect to state with parameters is not directly supported atm; I had another feature in mind — add support for `redirect: (ctx) => Promise|String` which is more generic and would solve your problem as well:

```es6
app.add('root', {
  path: '/',
  redirect: (ctx) => {
    ctx.params.id = 1;
    return 'form';
  }
});
```

This way it is a bit more uniform with `enter` hook, which, in turn, also allows modifying `params` by assigning them to "current" context.

**Comment by inca** [comment]
Hm, on second thought, the feature I've mentioned doesn't seem to be feasible: `redirect` on state can't accept `ctx`, because it would be performed before any transition takes place (and, therefore, no contexts yet created). This behavior is intentionally different from redirect via `enter` because it doesn't perform potentially unnecessary transitions.

So I guess I'll implement `redirect: { name, params }` as you've suggested.

**Comment by inca** [comment]
Btw, the problem you've mentioned could also be solved by making `id` param optional with default value:

```es6
app.add('form', {
  path: 'form/:id',
  params: {
    id: 1
  }
});
```

Proper support for defaults was added in 7a04d57 (release 0.6.2)

**Comment by inca** [comment]
Support for `redirect: { name, params }` is now available since 0.6.3. Cheers!
